src/hooks/useUserProfile.jsx

```diff
@@ -3,12 +3,11 @@ import { useEffect } from "react";
 
 import { useUserStore } from "@/store/useUserStore";
 
-const useUserProfile = () => {
+const useUserProfile = (userId) => {
   const { setUserSettings } = useUserStore();
 
   useEffect(() => {
```

Comment:
```suggestion
  useEffect(() => {
  if (!userId) {
    return;
  }
```
`useEffect`진입 초반에 early return으로 처리하면 함수 선언을 생략할 수 있어 코드가 조금 더 간결하고 의도가 명확해질 것 같습니다 :) ❕